projects/sandbox/analyze/analyze.py

```diff
@@ -183,111 +196,65 @@ def build_background(
                 else:
                     normalizer = None
 
-                # submit the integration job and have it update the
+                try:
+                    # if we already have a background distribution
+                    # for this normalization, grab it and fit it with a
+                    # "warm start" aka don't ditch the existing histogram
+                    background = backgrounds[norm]
+                    warm_start = True
+                except KeyError:
+                    # otherwise create a new distribution
+                    # and fit it from scratch
+                    background = ClusterDistribution("integrated", t_clust)
+                    backgrounds[norm] = background
+                    warm_start = False
+
+                # run integration in main process job and have it update the
                 # corresponding progress bar task once it completes
-                future = process_ex.submit(
-                    integrate,
+                t, y, integrated = integrate(
                     y,
                     t,
-                    kernel_length=1.0,
+                    kernel_length=kernel_length,
                     window_length=window_length,
                     normalizer=normalizer,
                 )
-                future.add_done_callback(
-                    lambda f: pbar.update(analyze_task_id, advance=1)
+
+                pbar.update(analyze_task_id, advance=1)
+
+                # fit background to integrated output
+                # in the main process
+                background.fit((integrated, t), warm_start=warm_start)
+
+                # submit the writing job to our thread pool and
+                # use a callback to keep track of all the filenames
+                # for a given normalization window
+                shift_dir = get_write_dir(
+                    write_dir, shift, "background-integrated", norm
                 )
-                integration_futures[(norm, shift)] = [future]
 
-            # advance the task keeping track of how many files
-            # we've loaded by one
-            pbar.update(load_task_id, advance=1)
+                future = thread_ex.submit(
```

Comment:
I feel like when I've tried to use a thread pool for reading/writing in the past I haven't seen particularly good performance, but it's been a while. Have you observed that this is able to work pretty well?

Reply:
So over the 10 timeslides the performance is basically the same (~ 40 seconds total for end to end analysis). I'm not sure how this will scale as we start analyzing more timeslides.

Reply:
The same as in it doesn't matter whether you use the thread or process pool for this?

Reply:
Yes

Reply:
Got it ok sounds good